driver/gl/window.go

```diff
@@ -503,50 +593,104 @@ func keyToName(key glfw.Key) fyne.KeyName {
 		return desktop.KeySuper
 	case glfw.KeyMenu:
 		return desktop.KeyMenu
-
-	case glfw.KeyKPEnter:
-		return fyne.KeyEnter
 	}
 	return ""
 }
 
 func (w *window) keyPressed(viewport *glfw.Window, key glfw.Key, scancode int, action glfw.Action, mods glfw.ModifierKey) {
-	if w.canvas.Focused() == nil && w.canvas.onTypedKey == nil {
-		return
-	}
+
 	if action != glfw.Press { // ignore key up
+		// TODO: handle repeated key
 		return
 	}
 
-	if key <= glfw.KeyWorld1 { // filter printable characters handled in charModInput
+	keyName := keyToName(key)
+	keyDesktopModifier := desktopModifier(mods)
+
+	if keyDesktopModifier < desktop.ControlModifier {
+		// No shortcut detected, pass down to TypedKey and exit
+		keyEvent := &fyne.KeyEvent{Name: keyName}
+		if w.canvas.Focused() != nil {
+			go w.canvas.Focused().TypedKey(keyEvent)
+		}
+		if w.canvas.onTypedKey != nil {
+			go w.canvas.onTypedKey(keyEvent)
+		}
 		return
 	}
 
-	ev := new(fyne.KeyEvent)
-	ev.Name = keyToName(key)
+	isMac := runtime.GOOS == "darwin"
 
-	if w.canvas.Focused() != nil {
-		go w.canvas.Focused().TypedKey(ev)
+	var shortcut fyne.Shortcuter
+	switch keyName {
+	case fyne.KeyV:
+		// detect paste shortcut
+		if (!isMac && keyDesktopModifier == desktop.ControlModifier) || (isMac && keyDesktopModifier == desktop.SuperModifier) {
+			shortcut = &fyne.ShortcutPaste{
+				Clipboard: w.Clipboard(),
+			}
+		}
+	case fyne.KeyC:
+		// detect copy shortcut
+		if (!isMac && keyDesktopModifier == desktop.ControlModifier) || (isMac && keyDesktopModifier == desktop.SuperModifier) {
+			shortcut = &fyne.ShortcutCopy{
+				Clipboard: w.Clipboard(),
+			}
+		}
+	case fyne.KeyX:
+		// detect cut shortcut
+		if (!isMac && keyDesktopModifier == desktop.ControlModifier) || (isMac && keyDesktopModifier == desktop.SuperModifier) {
+			shortcut = &fyne.ShortcutCut{
+				Clipboard: w.Clipboard(),
+			}
+		}
+	default:
+		shortcut = &desktop.CustomShortcut{
+			KeyName:  keyName,
+			Modifier: keyDesktopModifier,
+		}
+	}
+
+	if shortcutable, ok := w.canvas.Focused().(fyne.Shortcutable); ok {
+		go shortcutable.HandleShortcut(shortcut)
+	}
+
+	go w.canvas.HandleShortcut(shortcut)
```

Comment:
This should not be called if the shortcutable above was called and handled.
(see comment above about returning a bool)